tests/test_fmi_master.py

```diff
@@ -157,6 +167,46 @@ def test_basic_simulation_mat_file(self):
         opts = {"result_handling":"binary"}
         self._basic_simulation(opts)
     
+    @testattr(stddist = True)
+    def test_basic_simulation_mat_file_naming(self):
+        from pyfmi.common.algorithm_drivers import UnrecognizedOptionError
+        
+        opts = {"result_handling":"binary", "result_file_name": "Should fail..."}
+        
+        try:
+            self._basic_simulation(opts)
+        except UnrecognizedOptionError:
+            pass
+        
+        opts = {"result_handling":"binary", "result_file_name": {"Should fail..."}}
+        
+        try:
+            self._basic_simulation(opts)
+        except UnrecognizedOptionError:
+            pass
+    
+    @testattr(stddist = True)
+    def test_basic_simulation_mat_file_naming_exists(self):
+        models, connections = self._load_basic_simulation()
+        
+        opts = {"result_handling":"binary", "result_file_name": {models[0]: "Test1.mat", models[1]: "Test2.mat"}}
+        
+        res = self._sim_basic_simulation(models, connections, opts)
+        
+        assert os.path.isfile("Test1.mat"), "Test1.mat does not exists"
+        assert os.path.isfile("Test2.mat"), "Test2.mat does not exists"
+    
+    @testattr(stddist = True)
+    def test_basic_simulation_txt_file_naming_exists(self):
+        models, connections = self._load_basic_simulation()
+        
+        opts = {"result_handling":"file", "result_file_name": {models[0]: "Test1.txt", models[1]: "Test2.txt"}}
+        
+        res = self._sim_basic_simulation(models, connections, opts)
+        
+        assert os.path.isfile("Test1.txt"), "Test1.txt does not exists"
+        assert os.path.isfile("Test2.txt"), "Test2.txt does not exists"
+        
```

Comment:
I miss tests where result_handling is set to binary but result_file_name are specified as *.txt and vice versa.

Reply:
There is actually no restrictions on the suffix when the user provides a filename for the result. It is ok to provide .txt for binary files and .mat for text files. So no need to add a test.